libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpAmrReader.java

```diff
@@ -0,0 +1,193 @@
+   * band.
+   */
+  private static final int[] frameSizeBytesByTypeWb = {
+    18,
+    24,
+    33,
+    37,
+    41,
+    47,
+    51,
+    59,
+    61,
+    6, // AMR-WB SID
+    1, // Future use
+    1, // Future use
+    1, // Future use
+    1, // Future use
+    1, // speech lost
+    1 // No data
+  };
+
+  private final RtpPayloadFormat payloadFormat;
+  private @MonotonicNonNull TrackOutput trackOutput;
+  private long firstReceivedTimestamp;
+  private int previousSequenceNumber;
+  private long startTimeOffsetUs;
+  private final int sampleRate;
+  private boolean isWideBand;
+
+  public RtpAmrReader(RtpPayloadFormat payloadFormat) {
+    this.payloadFormat = payloadFormat;
+    firstReceivedTimestamp = C.TIME_UNSET;
+    previousSequenceNumber = C.INDEX_UNSET;
+    this.sampleRate = this.payloadFormat.clockRate;
+    this.isWideBand = (payloadFormat.format.sampleMimeType == MimeTypes.AUDIO_AMR_WB);
+  }
+
+  // RtpPayloadReader implementation.
+  @Override
+  public void createTracks(ExtractorOutput extractorOutput, int trackId) {
+    trackOutput = extractorOutput.track(trackId, C.TRACK_TYPE_AUDIO);
+    trackOutput.format(payloadFormat.format);
+  }
+
+  @Override
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {
+    this.firstReceivedTimestamp = timestamp;
+  }
+
+  @Override
+  public void consume(
+      ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker) {
+    // Check that this packet is in the sequence of the previous packet.
+    if (previousSequenceNumber != C.INDEX_UNSET) {
+      int expectedSequenceNumber = RtpPacket.getNextSequenceNumber(previousSequenceNumber);
+      if (sequenceNumber != expectedSequenceNumber) {
+        Log.w(
+            TAG,
+            Util.formatInvariant(
+                "Received RTP packet with unexpected sequence number. Expected: %d; received: %d.",
+                expectedSequenceNumber, sequenceNumber));
+      }
+    }
+    checkNotNull(trackOutput);
+    /**
+     * AMR as RTP payload RFC4867 Section-4.2
+     * +----------------+-------------------+----------------
+     * | payload header | table of contents | speech data ...
+     * +----------------+-------------------+----------------
+     *
+     * Payload header RFC4867 Section-4.4.1
+     * As interleaving is not supported currently, our header won't contain ILL and ILP
+     * +-+-+-+-+-+-+-+
+     * | CMR |R|R|R|R|
+     * +-+-+-+-+-+-+-+
+     */
+    // skip CMR and reserved bits
+    data.skipBytes(1);
+    // Loop over sampleSize to send multiple frames along with appropriate timestamp when compound
+    // payload support is added
+    int frameType = (data.peekUnsignedByte() >> 3) & 0x0f;
+    int frameSize = getFrameSize(frameType, isWideBand);
+    int sampleSize = data.bytesLeft();
+    checkArgument(sampleSize == frameSize, "compound payload not supported currently");
```

Comment:
Again, why is it not supported?

Reply:
We haven't been able to get an compound payload input from the live555 server. Since this feature is not being tested, we are not supporting it currently.